index.bs

```diff
@@ -1129,13 +1138,68 @@ a numbered step. If outdented, it (today) is rendered as a bullet in the midst o
 
                   1. If <code>|options|.{{PublicKeyCredentialCreationOptions/authenticatorSelection}}.{{authenticatorAttachment}}</code> is
                     [=present|present=] and its value is not equal to |authenticator|'s [=authenticator attachment modality=], [=iteration/continue=].
-                  1. If <code>|options|.{{PublicKeyCredentialCreationOptions/authenticatorSelection}}.{{requireResidentKey}}</code> is set to
-                    [TRUE] and the |authenticator| is not capable of storing a [=client-side-resident public key credential source=],
-                    [=iteration/continue=].
+
+                  1. If <code>|options|.{{PublicKeyCredentialCreationOptions/authenticatorSelection}}.{{residentKey}}</code>
+
+                      <dl class="switch">
+                          :   is set to {{ResidentKeyRequirement/required}}
+                          ::  If the |authenticator| is not capable of storing a [=client-side-resident public key credential
+                              source=], [=iteration/continue=].
+
+                          :   is set to {{ResidentKeyRequirement/preferred}}
+                          ::  If the |authenticator| is not capable of storing a [=client-side-resident public key credential
+                              source=], and there are other known authenticators attached that are capable of storing a
+                              [=client-side-resident public key credential source=], [=iteration/continue=].
+
+                          :   is set to {{ResidentKeyRequirement/discouraged}}
+                          ::  If the |authenticator| is not capable of creating a [=server-side-resident public key credential
```

Comment:
This wording is based around how the private key is stored, rather than whether the credential may be returned with an empty allowList. As a platform, we don't actually know how the private key is stored. For cost reasons, current authenticators tend to wrap it in the credential ID, but they might not.

Rather, `discouraged` to me is an opinion about that friction involved in registration. I see this as the RP saying &ldquo;please don't bother the user with a PIN prompt or hassle them about a fingerprint being rejected&rdquo;.

Reply:
Whether one wants PIN or not, should be controlled by userVerification. Although, currently, that is not mapping to intended behavior as Christiaan wants. Meaning, even if RP said that it does not want UV (UVdiscouraged), we end up having UV because of how CTAP is designed at this moment. 

Combining this definition with whether PIN prompt will be shown or not is not correct. May be in future, we may allow creating resident keys also without PIN prompt as Christiaan wanted. 

This parameter is about whether to create a resident key or not and we should only talk about its behavior in terms or that. 

From an Authenticator's perspective, they are given a freedom right now to enhance protection if they want. Meaning, in CTAP spec, even if request is not asking for UV, authenticator may do UV and that security decision is left to it and will be reflected in the flags.

Reply:
No action taken on this one based on comments from @akshayku